Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -432,13 +434,13 @@ static inline bool bounds_okay(const size_t cursor, const size_t delta, const si
 
 static inline bool is_valid_marker(const Marker marker)
 {
-    if (marker >= JPG_APPN0 && marker <= JPG_APPNF) {
-        if (marker != JPG_APPN0)
-            dbg() << String::format("%04x not supported yet. The decoder may fail!", marker);
+    // APPn markers are application dependant. Since interpretation is left to the application, these can be safely ignored.
+    if (marker >= JPG_APPN0 && marker <= JPG_APPNF)
         return true;
```

Comment:
The decoder was implemented with JFIF in mind. Non-JFIF images (APP1 or above) may pack information in different ways. For example, the assumption that component IDs won't exceed 3 comes from [JFIF](https://www.w3.org/Graphics/JPEG/jfif3.pdf). The only promise that JPEG makes about component ids is that they will be consistent across multiple scan headers and frame specs. The debug info is printed so that we know, in case of decoding failures, that the image not being a JFIF may be the reason why it failed. We don't give up decoding when we come across non-JFIF images because some interchange formats are perfectly compatible with JFIF and would not cause any trouble while decoding.